runtime/bcutil/cfreader.c

```diff
@@ -1635,6 +1635,21 @@ checkFields(J9PortLibrary* portLib, J9CfrClassFile * classfile, U_8 * segment, U
 			}
 		}
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		if (J9_ARE_ALL_BITS_SET(classfile->accessFlags, CFR_ACC_VALUE_TYPE)) {
+			if (J9_ARE_NO_BITS_SET(value, CFR_ACC_STATIC | CFR_ACC_FINAL)) {
+				errorCode = J9NLS_CFR_ERR_VALUE_CLASS_FIELD_NOT_STATIC_OR_FINAL__ID;
+				goto _errorFound;
+			}
+		}
+		if (J9_ARE_ALL_BITS_SET(classfile->accessFlags, CFR_ACC_ABSTRACT | CFR_ACC_PERMITS_VALUE)) {
```

Comment:
Do you need to check both `CFR_ACC_ABSTRACT` and `CFR_ACC_PERMITS_VALUE` ? If `CFR_ACC_PERMITS_VALUE` is set, `CFR_ACC_ABSTRACT` must also be set, otherwise it will return from error case here:
https://github.com/eclipse-openj9/openj9/blob/194ae711901aeec136e9679c660c2b6a9be086cb/runtime/bcutil/cfreader.c#L2907-L2909